### DISCUSSION TITLE
v0.3.0

### DISCUSSION THREAD

**Discussion by RobertCraigie** [RobertCraigie/prisma-client-py#113] [General]
# Breaking Changes

The `prisma` field name is now reserved, trying to generate a model that has a field called `prisma` will raise an error.

You can, however, still create a model that uses the `prisma` field name at the database level.

```prisma
model User {
  id           String @id @default(cuid())
  prisma_field String @map("prisma")
}
```

# What's Changed

## Querying from Model Classes

You can now run prisma queries directly from model classes, for example:

```py
from prisma.models import User

user = await User.prisma().create(
    data={
        'name': 'Robert',
    },
)
```

This API is *exactly* the same as the previous client-based API.

To get starting running queries from model classes, you must first register the prisma client instance that will be used to communicate with the database.

```py
from prisma import Client, register

client = Client()
register(client)
await client.connect()
```

For more details, visit the [documentation](https://prisma-client-py.readthedocs.io/en/latest/reference/model-actions/).

## Count non-null fields

You can now select which fields are returned by `count()`.

This returns a dictionary matching the fields that are passed in the `select` argument.

```py
from prisma.models import Post

results = await Post.prisma().count(
    select={
        '_all': True,
        'description': True,
    },
)
# {'_all': 3, 'description': 2}
```

## Support for Python 3.10

Python [3.10](https://docs.python.org/3/whatsnew/3.10.html) is now officially supported.

## Prisma Update

The internal Prisma binaries that Prisma Client Python uses have been upgraded from `3.3.0` to `3.4.0`.

- Support for PostgreSQL 14
- `prisma db push` support for MongoDB

## Improved Client Generation Message

The current version of the client will now be displayed post-generation:

```
Prisma schema loaded from schema.prisma

✔ Generated Prisma Client Python (v0.3.0) to ./.venv/lib/python3.9/site-packages/prisma in 765ms
```

## Improved Generation Error Message

An explicit and helpful message is now shown when attempting to generate the Python Client using an unexpected version of Prisma.

```
Environment variables loaded from .env
Prisma schema loaded from tests/data/schema.prisma
Error: 
  Prisma Client Python expected Prisma version: 1c9fdaa9e2319b814822d6dbfd0a69e1fcc13a85 but got: da6fafb57b24e0b61ca20960c64e2d41f9e8cff1
  If this is intentional, set the PRISMA_PY_DEBUG_GENERATOR environment variable to 1 and try again.
  Are you sure you are generating the client using the python CLI?
  e.g. python3 -m prisma generate (type=value_error)
```

## Other Changes

- added `--type-depth` option to `prisma py generate`

<hr /><em>This discussion was created from the release <a href='https://github.com/RobertCraigie/prisma-client-py/releases/tag/v0.3.0'>v0.3.0</a>.</em>

**Comment by rayli09** [comment]
getting same error

**Reply by RobertCraigie** [reply]
What error are you getting?
